go/pserver/service.go

```diff
@@ -49,13 +53,58 @@ type Service struct {
 	optMap map[string]*optimizer
 }
 
+// Checkpoint saves the checkpoint for pserver
+type Checkpoint struct {
+	UUID      string `json:"uuid"`
+	MD5       string `json:"md5"`
+	Timestamp string `json:"timestamp"`
+	State     []byte
```

Comment:
`Checkpoint` will be initialized from the JSON data in etcd, which only the the key `uuid`, `md5` and `timestamp`, but according with your other comment, separate `Checkpoint` and `CheckpointInfo` is better way, thanks for @helinwang .